SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py

```diff
@@ -0,0 +1,856 @@
+            var_att[self.sheet[self.idx_start_var_def, 5]] = self.sheet[i, 5]
+            var_def[self.sheet[i, 0]] = var_att
+        return var_def
+
+    def data_frame(self):
+        """ return the data as a pandas data frame """
+        dates = []
+        for i in range(self.idx_start_data + 1, self.idx_end_data):
+            dates.append(self.sheet[i, 0])
+
+        data  = {}
+        for j in range(1, self.sheet.number_of_columns()):
+            var_val = []
+            for i in range(self.idx_start_data + 1, self.idx_end_data):
+                var_val.append(self.sheet[i, j])
+            data[self.sheet[self.idx_start_data, j]] = var_val
+
+        data_frame = pd.DataFrame(data, index=pd.to_datetime(dates))
+        var_def    = self.dic_var_def()
+
+        if not all([col in var_def.keys() for col in data_frame.columns]):
+            if '' in data_frame.columns:
+                raise Exception('Empty column in middle of data. Please fix excel file %s' % self.filename)
+            else:
+                raise Exception('Not all variables are defined in TABLE_COLUMNS section. Please fix excel file %s' % self.filename)
+        else:
+            return data_frame
+
+class ReadXlsAbsorptionAC9HS6:
+    """ Read XLS Absorption and AC9_HS6 file """
+
+    def __init__(self, filename, sheetname=''):
+        self.filename=filename
+        self.sheetname=sheetname
+        if self.has_sheet():
+            self.sheet = pyexcel.get_sheet(file_name=filename, sheet_name=sheetname)
+            self.get_index_start_var_cols_def()
+            self.get_index_start_var_rows_def()
+            self.get_index_end_gatts()
+            self.get_index_start_data()
+            self.get_index_end_var_rows_def()
+            self.get_index_end_var_cols_def()
+            self.idx_end_data = self.sheet.number_of_rows()
+            self.max_data_column()
+
+    def has_sheet(self):
+        if self.sheetname == '':
+            return True
+
+        try:
+            pyexcel.get_sheet(file_name=self.filename, sheet_name=self.sheetname)
+            return True
+        except:
+            return False
+
+    def get_index_start_var_rows_def(self):
+        for i in range(self.sheet.number_of_rows()):
+            if self.sheet[i, 0] == "TABLE ROWS":
+                break
+        self.idx_start_var_rows_def = i + 1
+
+    def get_index_start_var_cols_def(self):
+        for i in range(self.sheet.number_of_rows()):
+            if self.sheet[i, 0] == "TABLE COLUMNS":
+                break
+        self.idx_start_var_cols_def = i + 1
+
+    def get_index_end_gatts(self):
+        self.idx_end_gatts = self.idx_start_var_rows_def-1
+
+    def get_index_start_data(self):
+        for i in range(self.sheet.number_of_rows()):
+            if self.sheet[i, 0] == "DATA":
+                break
+        self.idx_start_data = i + 1
+
+    def get_index_end_var_cols_def(self):
+        self.idx_end_var_cols_def = self.idx_start_data - 1
```

Comment:
it was to be consistent with other definitions, and made it easier to write code in a similar way